File: solution/3700-3799/3737.Count Subarrays With Majority Element I/Solution.java
class Solution {
    public int countMajoritySubarrays(int[] nums, int target) {
        int n = nums.length;
        int ans = 0;
        Map<Integer, Integer> cnt = new HashMap<>(n);
        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j) {
                int k = j - i + 1;
                cnt.merge(nums[j], 1, Integer::sum);
                if (cnt.getOrDefault(target, 0) > k / 2) {
                    ++ans;
                }
            }
            cnt.clear();
        }
        return ans;
    }
}

File: solution/3700-3799/3737.Count Subarrays With Majority Element I/Solution.ts
function countMajoritySubarrays(nums: number[], target: number): number {
    const n = nums.length;
    let ans = 0;
    for (let i = 0; i < n; ++i) {
        let cnt: number = 0;
        for (let j = i; j < n; ++j) {
            const k = j - i + 1;
            cnt += nums[j] == target ? 1 : 0;
            if (cnt * 2 > k) {
                ++ans;
            }
        }
    }
    return ans;
}
